integration/query_test.go

```diff
@@ -426,6 +427,202 @@ func TestQueryBadSortType(t *testing.T) {
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", math.MaxInt64},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "9223372036854775807 value for maxTimeMS is out of range",
+			},
+			altMessage: "9223372036854775807 value for maxTimeMS is out of range",
+		},
+		"BadMaxTimeMSMinInt64": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", math.MinInt64},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "-9223372036854775808 value for maxTimeMS is out of range",
+			},
+			altMessage: "-9223372036854775808 value for maxTimeMS is out of range",
+		},
+		"BadMaxTimeMSNull": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", nil},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "maxTimeMS must be a number",
+			},
+			altMessage: "maxTimeMS must be a number",
+		},
+		"BadMaxTimeMSArray": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", bson.A{int32(42), "foo", nil}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "maxTimeMS must be a number",
+			},
+			altMessage: "maxTimeMS must be a number",
+		},
+		"BadMaxTimeMSDocument": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", bson.D{{"foo", int32(42)}, {"42", "foo"}, {"array", bson.A{int32(42), "foo", nil}}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "maxTimeMS must be a number",
+			},
+			altMessage: "maxTimeMS must be a number",
+		},
+		"BadMaxTimeMSTypeNegativeInt32": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"projection", bson.D{{"value", "some"}}},
+				{"maxTimeMS", -1123123},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "-1123123 value for maxTimeMS is out of range",
+			},
+			altMessage: "-1123123 value for maxTimeMS is out of range",
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual bson.D
+			err := collection.Database().RunCommand(ctx, tc.command).Decode(&actual)
+			require.Error(t, err)
+			AssertEqualAltError(t, *tc.err, tc.altMessage, err)
+		})
+	}
+}
+
+func TestQueryMaxTimeMSAvailableValues(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+	for name, tc := range map[string]struct {
+		command any
+	}{
+		"MaxTimeMSZero": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"maxTimeMS", 0},
+			},
+		},
+		"MaxTimeMSInt32": {
+			command: bson.D{
+				{"find", collection.Name()},
+				{"maxTimeMS", 10000},
+			},
+		},
```

Comment:
Please add test case:
```
		"MaxTimeMSInt64": {
			command: bson.D{
				{"find", collection.Name()},
				{"maxTimeMS", int64(42)},
			},
		}
```

Reply:
I have `MaxTimeMSInt32` test case with `10000` value and it will be no difference with int64(42) because this value in the same range as int32. For int64 I have `BadMaxTimeMSMaxInt64` test case

Reply:
Agree, added